src/napari_matplotlib/line.py

```diff
@@ -0,0 +1,246 @@
+        x_axis_name : str
+            The label to display on the x axis
+        y_axis_name: str
+            The label to display on the y axis
+        """
+        raise NotImplementedError
+
+class MetadataLine2DWidget(Line2DBaseWidget):
+    n_layers_input = Interval(1, 1)
+
+    def __init__(self, napari_viewer: napari.viewer.Viewer):
+        super().__init__(napari_viewer)
+        self.setMinimumSize(200, 200)
+        self._plugin_name_widget = magicgui(
+            self._set_plugin_name,
+            plugin_name={"choices": self._get_plugin_metadata_key},
+            auto_call = True,
+        )
+        self._key_selection_widget = magicgui(
+            self._set_axis_keys,
+            x_axis_key={"choices": self._get_valid_axis_keys},
+            y_axis_key={"choices": self._get_valid_axis_keys},
+            call_button="plot",
+        )
+        
+        self.ray_index = 0
+        self.dot_index = 0
+        self.layout().insertWidget(0, self._plugin_name_widget.native)
+        self.layout().addWidget(self._key_selection_widget.native)
+
+    @property
+    def x_axis_key(self) -> Optional[str]:
+        """Key to access x axis data from the Metadata"""
+        return self._x_axis_key
+
+    @x_axis_key.setter
+    def x_axis_key(self, key: Optional[str]) -> None:
+        self._x_axis_key = key
+        self._draw()
+
+    @property
+    def y_axis_key(self) -> Optional[str]:
+        """Key to access y axis data from the Metadata"""
+        return self._y_axis_key
+
+    @y_axis_key.setter
+    def y_axis_key(self, key: Optional[str]) -> None:
+        self._y_axis_key = key
+        self._draw()
+
+    def _set_axis_keys(self, x_axis_key: str, y_axis_key: str) -> None:
+        """Set both axis keys and then redraw the plot"""
+        self._x_axis_key = x_axis_key
+        self._y_axis_key = y_axis_key
+        self._draw()
+        
+    @property
+    def plugin_name_key(self) -> Optional[str]:
+        """Key to plugin dictionary in the Metadata"""
+        return self._plugin_name_key
+
+    @plugin_name_key.setter
+    def plugin_name_key(self, key: Optional[str]) -> None:
+        self._plugin_name_key = key
+        
+    def _set_plugin_name(self, plugin_name: str) -> None:
+        """Set plugin name from layer metadata"""
+        self._plugin_name_key = plugin_name
+        self._key_selection_widget.reset_choices()
+    
+    def _get_plugin_metadata_key(
+            self, combo_widget: Optional[ComboBox] = None
+        ) -> List[str]:
+        """Get plugin key from layer metadata"""
+        if len(self.layers) == 0:
+            return []
+        else:
+            return self._get_valid_metadata_keys() 
+
+    def _get_valid_metadata_keys(
+            self) -> List[str]:
+        """Get metadata keys if nested dictionaries"""
+        if len(self.layers) == 0:
+            return []
+        else:
+            metadata = self.layers[0].metadata
```

Comment:
Would it be possible to get the metadata from all the layers? That way if more than one layer was selected you could choose from metadata in any of the selected layers.

Reply:
That is a good idea!